Zip archive handling

atomicguy opened the pull request:
Proposed solution for importing zipped folders of audio files

GianniCarlo commented on BookPlayer/Library/FileManagement/ImportOperation.swift:
```diff
@@ -30,6 +31,52 @@ class ImportOperation: Operation {
+                    }
+                }
+            }
+
```
we should open the file stream here. It's being open on line 28, but not being used if it's a zip file

GianniCarlo commented on BookPlayer/Library/FileManagement/ImportOperation.swift:
```diff
@@ -21,15 +22,67 @@ class ImportOperation: Operation {
+                try FileManager.default.unzipItem(at: file.originalUrl, to: tempURL)
+
+            } catch {
+                print("Extraction of ZIP archive failed with error:\(error)")
```
we should add a `return` here, if it fails to unzip the file the rest shouldn't be executed

GianniCarlo commented on BookPlayer/Library/FileManagement/ImportOperation.swift:
```diff
@@ -21,15 +22,67 @@ class ImportOperation: Operation {
+                print("Extraction of ZIP archive failed with error:\(error)")
+            }
+
+            do {
```
This `do` doesn't have a `catch`, and everything else is already in a inner `do`, we can safely delete this

GianniCarlo commented on BookPlayer/Library/FileManagement/ImportOperation.swift:
```diff
@@ -21,15 +22,67 @@ class ImportOperation: Operation {
+                            try FileManager.default.moveItem(at: fileURL, to: destinationURL)
+                        }
+                    }
+                } catch {
```
since we're not doing anything in this `catch` we can remove the `do` `catch` and let `FileManager.default.moveItem` fail silently using `try? FileManager.default.moveItem `

GianniCarlo commented on BookPlayer/Library/FileManagement/ImportOperation.swift:
```diff
@@ -21,15 +22,67 @@ class ImportOperation: Operation {
+                // Delete temp directory
+                do {
+                    try FileManager.default.removeItem(at: tempURL)
+                } catch {
```
same thing here, since no error handling is happening, use `try? ` to let it fail silently

GianniCarlo commented:
to fix the tests of travis, you've got to add the `ZIPFoundation` framework in both the *Input Files* and *Output Files* sections in the Build Phases tab

<img width="691" alt="screen shot 2018-10-06 at 2 26 24 pm" src="https://user-images.githubusercontent.com/14112819/46575062-d06c9880-c973-11e8-9b55-479bb6d693aa.png">

And lastly, this PR should be made pointing to the `develop` branch instead of `master

GianniCarlo commented:
with the tests fixed, and the PR pointing to `develop` we can go ahead with the merge 🎉

atomicguy commented:
Fixed where the PR is pointing.

![screen shot 2018-10-06 at 9 47 59 pm](https://user-images.githubusercontent.com/395176/46575223-b05bca00-c9b1-11e8-8fec-de76855a2e50.png)
I've already added ZipFoundation to _Input_ and _Output_ Files inside the Carthege section. I'm not sure where else to add it.

GianniCarlo commented:
oh right 🤔, mmmmm let me see what else could be the cause of travis not finding it

atomicguy commented:
Looks like I needed to add the updated Cartfile.resolved 🤐

GianniCarlo commented:
Thanks for the contribution @atomicguy 🎉
